syntax = "proto2";

package NCloud.NFileStore.NProto;

import "cloud/filestore/config/client.proto";

option go_package = "a.yandex-team.ru/cloud/filestore/config";

////////////////////////////////////////////////////////////////////////////////

message TServerConfig
{
    // Host name or address to listen on.
    optional string Host = 1;

    // Port to listen on.
    optional uint32 Port = 2;

    // Number of gRPC threads.
    optional uint32 ThreadsCount =3;

    // Limit of grpc threads count.
    optional uint32 GrpcThreadsLimit = 4;

    // Maximum size of gRPC message.
    optional uint32 MaxMessageSize = 5;

    // Memory quota for gRPC request processing.
    optional uint32 MemoryQuotaBytes = 6;

    // Number of prepared gRPC requests.
    optional uint32 PreparedRequestsCount = 7;

    // KeepAlive details.
    optional bool KeepAliveEnabled = 8;
    optional uint32 KeepAliveIdleTimeout = 9;   // (in milliseconds).
    optional uint32 KeepAliveProbeTimeout = 10; // (in milliseconds).
    optional uint32 KeepAliveProbesCount = 11;

    optional uint32 ShutdownTimeout = 12; // (in milliseconds).
}

////////////////////////////////////////////////////////////////////////////////

message TNullServiceConfig
{
}

////////////////////////////////////////////////////////////////////////////////

message TLocalServiceConfig
{
    optional string RootPath = 1;
    optional string PathPrefix = 2;
    optional uint32 DefaultPermissions = 3;
    optional uint32 IdleSessionTimeout = 4;
    optional uint32 NumThreads = 5;
}

////////////////////////////////////////////////////////////////////////////////

message TKikimrServiceConfig
{
}

////////////////////////////////////////////////////////////////////////////////

message TServerAppConfig
{
    optional TServerConfig ServerConfig = 1;

    optional TNullServiceConfig NullServiceConfig = 2;
    optional TLocalServiceConfig LocalServiceConfig = 3;
    optional TKikimrServiceConfig KikimrServiceConfig = 4;
}
